config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,49 @@
+RACK_ATTACK_THROTTLE_PATHS = %w(/users /users/sign_in /users/unlock).freeze
+
+RACK_ATTACK_CONNECTION_POOL = ConnectionPool.new(size: 10, timeout: 0.1) do
+  redis = Redis.new(Shuttle::Configuration.redis.symbolize_keys)
+  Redis::Namespace.new('rack::attack', redis: redis)
+end
+
+Rack::Attack.cache.store = RateLimitRedisCache.new(RACK_ATTACK_CONNECTION_POOL)
+
+# Lockout IP addresses that are hammering user sign ups, sign ins and unlocks.
+Rack::Attack.blacklist('allow2ban lead form scrappers') do |req|
+  if RACK_ATTACK_THROTTLE_PATHS.include?(req.path) && req.post?
+    # This is used to experiment what the proper IP should be used in throttling.
+    ip = req.ip
+    client_ip = req.env['HTTP_CLIENT_IP']
+    forward_ip = req.env['HTTP_X_FORWARDED_FOR']
+    remote_addr = req.env['REMOTE_ADDR']
+    Rails.logger.info("RACK-ATTACK checking - ip:#{ip} client_ip:#{client_ip} forward_ip:#{forward_ip} remote_addr:#{remote_addr}")
+
+    # After 20 requests in 5 minute, block all requests from that IP for 1 hour.
+    real_ip = client_ip || remote_addr || forward_ip || ip || 'unknown ip'
+    Rack::Attack::Allow2Ban.filter(real_ip, maxretry: 20, findtime: 5.minutes, bantime: 1.hour) do
```

Comment:
i think this is fine, but maybe we should limit signups more? Do we imagine that someone's going to have to sign up more than an couple times in an hour? The only situation I can think of is if we onboard a bunch of new translators and they all sign up from the same computer at once. Even then, it seems unlikely.

Reply:
Currently the 20 counts are combined for all operations of sign up, unlock and sign in.